File: bancoDev/src/app/services/prestamo.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environment/environment';
import { ApiResponse } from '../interfaces/ApiResponse';
import { PrestamoResponse } from '../interfaces/prestamos/PrestamoResponse';
import { PrestamoCrearDto } from '../interfaces/prestamos/PrestamoCrearDto';
import { PrestamoCrearFinalDto } from '../interfaces/prestamos/PrestamoCrearFinalDto';

@Injectable({
  providedIn: 'root'
})
export class PrestamoService {

  private api:string = `${environment.url}/prestamo`;

  constructor( private http: HttpClient ) { }

  listarTodosLosPrestamos(idCliente:number): Observable<ApiResponse<PrestamoResponse[]>> {
    return this.http.get<ApiResponse<PrestamoResponse[]>>(`${this.api}/listarTodosLosPrestamos/${idCliente}`);
  }

  listarPrestamosPagados(idCliente:number): Observable<ApiResponse<PrestamoResponse[]>> {
    return this.http.get<ApiResponse<PrestamoResponse[]>>(`${this.api}/listarPrestamosPagados/${idCliente}`);
  }

  listarPrestamosPendientesEmpleado(idEmpleado:number): Observable<ApiResponse<PrestamoResponse[]>> {
    return this.http.get<ApiResponse<PrestamoResponse[]>>(`${this.api}/listarPrestamosPendientesEmpleado/${idEmpleado}`);
  }

  listarPrestamosPendientes(idCliente:number): Observable<ApiResponse<PrestamoResponse[]>> {
    return this.http.get<ApiResponse<PrestamoResponse[]>>(`${this.api}/listarPrestamosPendientes/${idCliente}`);
  }

  buscarPrestamosPorDniCliente(dni:string): Observable<ApiResponse<PrestamoResponse[]>> {
    return this.http.get<ApiResponse<PrestamoResponse[]>>(`${this.api}/buscarPrestamosPorDniCliente/${dni}`);
  }

  buscarPrestamo(id:number): Observable<ApiResponse<PrestamoResponse>> {
    return this.http.get<ApiResponse<PrestamoResponse>>(`${this.api}/buscarPrestamo/${id}`);
  }

  crearPrestamo(idPrestamo: number): Observable<ApiResponse<PrestamoResponse>> {
    return this.http.get<ApiResponse<PrestamoResponse>>(`${this.api}/crearPrestamo/${idPrestamo}`);
  }

  solicitarPrestamo(prestamo:PrestamoCrearDto): Observable<ApiResponse<PrestamoResponse>> {
    return this.http.post<ApiResponse<PrestamoResponse>>(`${this.api}/solicitarPrestamo`, prestamo);
  }

}

File: BancoDev - BackEnd/5 MicroservicioPrestamo/src/main/java/com/bancoDev/services/PrestamoService.java
package com.bancoDev.services;

import java.util.List;

import com.bancoDev.DTOs.ApiResponse;
import com.bancoDev.DTOs.request.PrestamoCrearDto;
import com.bancoDev.DTOs.response.PrestamoResponse;

public interface PrestamoService {

    ApiResponse<List<PrestamoResponse>> listarTodosLosPrestamos(Long idCliente);
    ApiResponse<List<PrestamoResponse>> listarPrestamosPagados(Long idCliente);
    ApiResponse<List<PrestamoResponse>> listarPrestamosPendientes(Long idCliente);
    ApiResponse<List<PrestamoResponse>> listarPrestamosPendientesEmpleado(Long idEmpleado);
    ApiResponse<List<PrestamoResponse>> buscarPrestamosPorDniCliente(String dni);
    ApiResponse<List<PrestamoResponse>> buscarPrestamosPorDniEmpleado(String dni);
    ApiResponse<PrestamoResponse> buscarPrestamo(Long id);
    ApiResponse<PrestamoResponse> crearPrestamo(Long id);
    ApiResponse<PrestamoResponse> solicitarPrestamo(PrestamoCrearDto prestamo);
    

}
